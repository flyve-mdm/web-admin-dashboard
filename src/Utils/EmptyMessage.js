import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Calc100PercentMinus from './Calc100PercentMinus'

class EmptyMessage extends Component {
    render () {
        return (
            <div className="contentPane" style={{ width: Calc100PercentMinus(this.props.itemListPaneWidth) }}>
                <div style={{ display: 'flex', height: '100%', justifyContent: 'center', alignItems: 'center', flexDirection: 'column' }}>
                    <h1 className="win-h1" style={{ color: 'grey' }}> 
                        { this.props.message } 
                    </h1>
                </div>
            </div>
        )
    }
}

EmptyMessage.propTypes = {
    itemListPaneWidth: PropTypes.number.isRequired,
    message: PropTypes.string.isRequired
}

export default EmptyMessage