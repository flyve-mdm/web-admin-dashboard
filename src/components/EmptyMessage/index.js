/*
*   Copyright © 2018 Teclib. All rights reserved.
*
*   This file is part of web-mdm-dashboard
*
* web-mdm-dashboard is a subproject of Flyve MDM. Flyve MDM is a mobile
* device management software.
*
* Flyve MDM is free software: you can redistribute it and/or
* modify it under the terms of the GNU General Public License
* as published by the Free Software Foundation; either version 3
* of the License, or (at your option) any later version.
*
* Flyve MDM is distributed in the hope that it will be useful,
* but WITHOUT ANY WARRANTY; without even the implied warranty of
* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
* GNU General Public License for more details.
* ------------------------------------------------------------------------------
* @author     Gianfranco Manganiello (gmanganiello@teclib.com)
* @author     Hector Rondon (hrondon@teclib.com)
* @copyright  Copyright © 2018 Teclib. All rights reserved.
* @license    GPLv3 https://www.gnu.org/licenses/gpl-3.0.html
* @link       https://github.com/flyve-mdm/web-mdm-dashboard
* @link       http://flyve.org/web-mdm-dashboard
* @link       https://flyve-mdm.com
* ------------------------------------------------------------------------------
*/

import React from 'react'
import PropTypes from 'prop-types'
import { I18n } from "react-i18nify"

/**
 * Component to show a message when the screen is void
 * @function EmptyMessage
 * @param {object} props
 * @return {component} Message to show
 */
const EmptyMessage = props => {
    /** Render component */
    return (
        <div className="center-block-content">
            {props.icon && props.showIcon ? <span className={"icon-empty-message " + props.icon} /> : null}
            <h1 className="empty-message"> 
                { props.message } 
            </h1>
        </div>
    )
}

EmptyMessage.propTypes = {
    icon: PropTypes.string,
    showIcon: PropTypes.bool,
    message: PropTypes.string.isRequired
}

EmptyMessage.defaultProps = {
    showIcon :false,
    message: I18n.t('commons.no_selection')
}

export default EmptyMessage