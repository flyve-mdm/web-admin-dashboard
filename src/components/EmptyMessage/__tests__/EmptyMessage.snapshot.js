import React from 'react'
import renderer from 'react-test-renderer'

import EmptyMessage from "../index"

describe('EmptyMessage', () => {
    test('renders EmptyMessage', () => {
        const component = renderer.create(
            <EmptyMessage />
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})