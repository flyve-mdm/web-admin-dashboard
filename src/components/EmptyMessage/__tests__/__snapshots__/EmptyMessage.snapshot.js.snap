// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyMessage renders EmptyMessage 1`] = `
<div
  className="center-block-content"
>
  <h1
    className="empty-message"
  >
    no_selection
  </h1>
</div>
`;
